test/engine/SpatialJoinAlgorithmsTest.cpp

```diff
@@ -780,8 +781,277 @@ TEST_P(SpatialJoinParamTest, computeResultSmallDatasetDifferentSizeChildren) {
+  for (size_t a = 0; a <= 1; a++) {
+    for (size_t b = 0; a <= 1; a++) {
+      for (size_t c = 0; a <= 1; a++) {
+        for (size_t d = 0; a <= 1; a++) {
+          for (size_t e = 0; a <= 1; a++) {
+            std::vector<box> toTest;  // the set of bounding boxes
+            std::vector<std::vector<point>> shouldBeContained;
+            std::vector<std::vector<point>> shouldNotBeContained;
+            if (a == 1) {  // box nr. 0 is contained in the set of boxes
+              toTest.push_back(boxes.at(0));
+              shouldBeContained.push_back(containedInBox.at(0));
+            } else {  // box nr. 0 is not contained in the set of boxes
+              shouldNotBeContained.push_back(containedInBox.at(0));
+            }
+            if (b == 1) {  // box nr. 1 is contained in the set of boxes
+              toTest.push_back(boxes.at(1));
+              shouldBeContained.push_back(containedInBox.at(1));
+            } else {  // box nr. 1 is not contained in the set of boxes
+              shouldNotBeContained.push_back(containedInBox.at(1));
+            }
+            if (c == 1) {
+              toTest.push_back(boxes.at(2));
+              shouldBeContained.push_back(containedInBox.at(2));
+            } else {
+              shouldNotBeContained.push_back(containedInBox.at(2));
+            }
+            if (d == 1) {
+              toTest.push_back(boxes.at(3));
+              shouldBeContained.push_back(containedInBox.at(3));
+            } else {
+              shouldNotBeContained.push_back(containedInBox.at(3));
+            }
+            if (e == 1) {
+              toTest.push_back(boxes.at(4));
+              shouldBeContained.push_back(containedInBox.at(4));
+            } else {
+              shouldNotBeContained.push_back(containedInBox.at(4));
+            }
+            if (toTest.size() > 0) {
+              // test all points, which should be contained in the bounding
+              // boxes
+              for (size_t i = 0; i < shouldBeContained.size(); i++) {
+                for (size_t k = 0; k < shouldBeContained.at(i).size(); k++) {
+                  ASSERT_TRUE(
+                      spatialJoinAlgs
+                          .OnlyForTestingWrapperContainedInBoundingBoxes(
+                              toTest, shouldBeContained.at(i).at(k)));
+                }
+              }
+              // test all points, which shouldn't be contained in the bounding
+              // boxes
+              for (size_t i = 0; i < shouldNotBeContained.size(); i++) {
+                for (size_t k = 0; k < shouldNotBeContained.at(i).size(); k++) {
+                  ASSERT_FALSE(
+                      spatialJoinAlgs
+                          .OnlyForTestingWrapperContainedInBoundingBoxes(
+                              toTest, shouldNotBeContained.at(i).at(k)));
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+}  // namespace boundingBox
```

Comment:
I didn't yet look at all those tests in detail, 
but something you could also do: 
Get a bounding box from your implementation, and get a bounding box via S2, and then
check that they are consistent etc.

Reply:
yes, that's a possibility. Although i think the current tests are already sufficient. If you want to i can implement it though.